code/game/objects/items.dm

```diff
@@ -207,7 +207,8 @@
 	if (istype(loc, /obj/item/weapon/storage))
 		//If the item is in a storage item, take it out.
 		var/obj/item/weapon/storage/S = loc
-		S.remove_from_storage(src, user)
+		if(!S.remove_from_storage(src, user))
+			return
```

Comment:
Adding a return here means it doesn't attempt put_in_active_hand later on down the process